Adds ability to resist out of gift-wrapped mob

emmanuelbassil opened the pull request:
🆑 emmanuelbassil
bugfix: Can now resist out of the human-sized gift.
tweak: Added more feedback to chewing self out of cuffs. Did not change behavior, need to target mouth on harm intent and click on self to do so.
tweak: Added more feedback to resisting out of closets; giving you progress towards success.
bugfix: Fixed being unable to break out of welded closets.
/🆑

SierraKomodo commented on code/game/objects/items/weapons/handcuffs.dm:
```diff
@@ -103,30 +103,38 @@
+		to_chat(src, SPAN_WARNING("You need to target your mouth to start chewing through your restraints!"))
+		return
+	if (istype(H.wear_suit, /obj/item/clothing/suit/straight_jacket))
+		to_chat(src, SPAN_WARNING("Try as you might, you cannot chew through your straight jacket."))
```
```suggestion
		to_chat(src, SPAN_WARNING("Try as you might, you cannot chew through your [H.wear_suit.name]."))
        return
```

SierraKomodo commented on code/game/objects/items/weapons/gifts.dm:
```diff
@@ -84,13 +84,15 @@
 		return INITIALIZE_HINT_QDEL
 
 	var/mob/living/carbon/human/human = target
+	human.handcuffed = src
```
`handcuffed` and the logic using it expect an item subtype. This violates that contract.

SierraKomodo reviewed (dismissed):
Over-all not a fan of this method. I'd recommend looking into how closets handle breaking out and do something similar to that instead - Or make the resist verb just damage the wrapping's health.

emmanuelbassil committed:
Resist out of gift mob

Fixes

Auto stash before rebase of "refs/heads/Resistwrap"

Auto stash before rebase of "refs/heads/Resistwrap"

Fixes

emmanuelbassil commented:
Fixed as requested; much more elegant now. 

Changed closet resist code to give user progress feedback similar to handcuffs.
Mobpresents are now a subtype of structure; and uses similar resist code to closets.
Renamed breakouttime to breakout_time
Fixed being unable to resist out of open; unlocked; unwelded closets. Clicking resist now simply opens the door.
Fixed oversight making it that you couldn't resist out of welded closets.

Updated changelog
